tests/chart_tests/test_byo_sa.py

```diff
@@ -75,6 +75,7 @@ def test_serviceaccount_with_overrides(self, kube_version):
         }
         extracted_names = {doc["metadata"]["name"] for doc in docs if "metadata" in doc and "name" in doc["metadata"]}
         assert expected_names.issubset(extracted_names)
+        assert all(doc["automountServiceAccountToken"] is True for doc in docs if doc["automountServiceAccountToken"])
```

Comment:
This does not do what you think it does. It will not include anything where `automountServiceAccountToken` is not `False`

```
>>> docs = [{'automountServiceAccountToken': True}, {'automountServiceAccountToken': False}]
>>> len([doc for doc in docs if doc["automountServiceAccountToken"]])
1
>>> len([doc for doc in docs if doc.get("automountServiceAccountToken")])
1
>>> all([doc["automountServiceAccountToken"] is True for doc in docs])
False
```

This test actually passes without that logic error though, so we can just remove the `if`

```suggestion
        assert all(doc["automountServiceAccountToken"] is True for doc in docs)
```